scipy/sparse/_base.py

```diff
@@ -340,18 +339,23 @@ def __str__(self):
         A = self.tocoo()
 
         # helper function, outputs "(i,j)  v"
-        def tostr(row, col, data):
-            triples = zip(list(zip(row, col)), data)
-            return '\n'.join([('  {}\t{}'.format(*t)) for t in triples])
+        def tostr(coords, data):
+            pairs = zip(zip(*coords), data)
+            return '\n'.join(f'  {idx}\t{val}' for idx, val in pairs)
```

Comment:
Note: in Numpy 2.0+, this will format the indices with `np.int32()` around each one. For example:

```
  (np.int32(3), np.int32(1))    0.2435
```